src/api/app/queries/event_subscriptions_finder.rb

```diff
@@ -3,11 +3,29 @@ def initialize(relation = EventSubscription.all)
     @relation = relation
   end
 
-  def for_scm_channel_with_token(event_type:, event_package:)
+  def for_scm_channel_with_token(event_type:, event_package_or_request:)
+    if event_package_or_request.is_a(Package)
+      find_for_a_package(event_type: event_type, event_package: event_package_or_request)
+    else
+      find_for_a_request(event_type: event_type, event_request: event_package_or_request)
+    end
+  end
+
+  private
+
+  def find_for_a_package(event_type:, event_package:)
     @relation
       .where(eventtype: event_type,
              package: event_package,
              channel: :scm)
       .where.not(token_id: nil)
   end
+
+  def find_for_a_request(event_type:, event_request:)
+    @relation
```

Comment:
I don't think you need to duplicate all of the query. You can just add `.where(package: event_package` or `.where(bs_request: event_request)` to `@relation` depending on which object you have.

Reply:
Changed